MainApp/lib/app/routes/api.js

```diff
@@ -29,9 +29,12 @@ export default function (router) {
         trialinfo.username = req.body.username;
         trialinfo.trialid = req.body.trialid;
         trialinfo.condition = req.body.condition;
-        
-        if (trialinfo.username == null || trialinfo.username == '' || trialinfo.trialid == null || trialinfo.trialid == '' || trialinfo.condition == null ||trialinfo.condition == '') {
-            res.send({success: false, message: 'Username or trialid or condition was empty' + trialinfo.username + ' ID:' +trialinfo.trialid + 'CON:' + trialinfo.condition});
+
+        if (trialinfo.username == null || trialinfo.username == '' || trialinfo.trialid == null || trialinfo.trialid == '' || trialinfo.condition == null || trialinfo.condition == '') {
```

Comment:
if(!trialinfo.trialid) when trialid=0 evaluates to false instead of true, so it not sufficient for the case of 0 when the input is a number. 
PS - This is actually not my code if you notice the diff, I must have auto-indented it or something. For the gameConfiguration parameters I need to have 0 as a valid input for number of doctors,nurses etc.